ludwig/features/boundingbox_feature.py

```diff
@@ -0,0 +1,333 @@
+
+        with tf.variable_scope('predictions_{}'.format(self.name)):
+            initializer_obj = get_initializer(self.initializer)
+            weights = tf.get_variable(
+                'weights',
+                initializer=initializer_obj([hidden_size, self.bounding_box_size]),
+                regularizer=regularizer
+            )
+            logging.debug('  regression_weights: {0}'.format(weights))
+
+            biases = tf.get_variable(
+                'biases', 
+                [self.bounding_box_size]
+            )
+            logging.debug('  regression_biases: {0}'.format(biases))
+
+            predictions = tf.reshape(
+                tf.matmul(hidden, weights) + biases,
+                [-1, 4]
+            )
+
+            predictions = tf.to_int32(predictions)
+
+            logging.debug('  predictions: {0}'.format(predictions))
+
+            if self.clip is not None:
+                if isinstance(self.clip, (list, tuple)) and len(self.clip) == 2:
+                    predictions = tf.clip_by_value(
+                        predictions,
+                        self.clip[0],
+                        self.clip[1]
+                    )
+                    logging.debug(
+                        '  clipped_predictions: {0}'.format(predictions)
+                    )
+                else:
+                    raise ValueError(
+                        'The clip parameter of {} is {}. '
+                        'It must be a list or a tuple of length 2.'.format(
+                            self.name,
+                            self.clip
+                        )
+                    )
+
+        return predictions
+
+    def _get_loss(self, targets, predictions):
+        with tf.variable_scope('loss_{}'.format(self.name)):
+            if self.loss['type'] == HUBER_LOSS:
+                train_loss = tf.losses.huber_loss(
+                    targets,
+                    predictions,
+                    loss_collection=None,
+                    reduction=Reduction.NONE)
+
+            elif self.loss['type'] == 'yolov3_loss': #to implement
+                pass
+            else:
+                train_mean_loss = None
+                train_loss = None
+                raise ValueError(
+                    'Unsupported loss type {}'.format(self.loss['type'])
+                )
+
+            train_mean_loss = tf.reduce_mean(
+                train_loss,
+                name='train_mean_loss_{}'.format(self.name)
+            )
+
+        return train_mean_loss, train_loss
+
+    def _get_measures(self, targets, predictions):
+        with tf.variable_scope('measures_{}'.format(self.name)):
+            error_val = get_error(
+                targets,
+                predictions,
+                self.name
+            )
+
+            iou_val = get_bbox_iou(
+                targets,
+                predictions,
+                self.name
+            )
+
+            mean_average_precision = None
```

Comment:
usually in ludwig we don't compute means in the graph because when we evaluate we want the means to be computed on the whole set rather than on the specific batch. There is a mechanism for collecting all the outputs of the batches when evaluating and then performa the average just one time. look at the code in model.py that uses OutputConfig for more details.